### ISSUE TITLE
Empty array with an annotation does not trigger the EmptyArrayAccess issue

### ISSUE THREAD

**Issue by LeSuisse** [vimeo/psalm#1706]
Result: No issue
Expected: EmptyArrayAccess - 5:1 - Cannot access value on empty array variable $ar
```php
<?php

/** @var array $ar */
$ar = [];
$ar['a'];
```
https://psalm.dev/r/c3e01da821

**Comment by muglug** [comment]
This is by design - `@var` overrides types

**Comment by LeSuisse** [comment]
Yup, sorry about that. It makes complete sense!
